### DISCUSSION TITLE
Is there a way to grab the parent argument in  a DgsData function?

### DISCUSSION THREAD

**Discussion by BCantos17** [Netflix/dgs-framework#1032] [Q&A] [answered]
Given a schema like this 

```graphql
type Actor {
    name: String
    movies(size: Int): [Movie]
}

type Movie {
    name: String
}

type Query {
    actor(id: String)
}
```

and given a movie field implementation. is there a way to grab the `id` field from the actor query? Maybe something like this?

```kotlin
    @DgsData(parentType = "Actor", field = "movies")
    fun movies(dfe: DgsDataFetchingEnvironment, @InputArgument size: Int): List<Movie> {
        val id = dfe.arguments["id"] as String? ?: ""
        return movieService.getMoviesByActorId(id, size)
    }

```

Currently `arguments` only has the the arguments for the query it is being called in and not the parent query

**Comment by srinivasankavitha** [comment, marked as answer]
You can pass any additional information as part of the context to child data fetchers, so you could add it there: https://netflix.github.io/dgs/advanced/context-passing/#no-showid-use-local-context

The other alternative for the parent is to inspect the Data fetching environment that's passed in and extract it, a bit more convoluted.

Hope this helps.

**Reply by BCantos17** [reply]
Does the first solution work with reactive code? Would I return a Mono with a DataFetcherResult or do I return a DataFetcherResult with a Mono?

**Reply by BCantos17** [reply]
Just tested it out, it works with a mono of a DataFetcherResult. Thanks @srinivasankavitha!

**Reply by srinivasankavitha** [reply]
👍 . Yes it works the same way with Mono as well.
